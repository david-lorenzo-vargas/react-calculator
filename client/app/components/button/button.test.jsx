import React from 'react';
import { shallow } from 'enzyme';
import Button from './button';

describe('Button', () => {
  describe('render', () => {
    it('should render correctly', () => {
      const props = [
        {
          type: 'num',
          value: 0,
          text: '0',
          theme: 'grey',
          size: 'big',
        },
        {
          type: 'num',
          value: 0,
          text: '0',
          theme: 'grey',
        },
        {
          type: 'num',
          value: 0,
          text: '0',
          theme: 'grey',
          color: 'pink',
        },
      ];

      props.forEach((propItem) => {
        const wrapper = shallow(<Button {...propItem} />);
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('when I click the button', () => {
      it('should call onClick function passing value and type', () => {
        const onClick = jest.fn();
        const wrapper = shallow(<Button
          type="num"
          value={0}
          text="0"
          theme="grey"
          color="pink"
          onClick={onClick}
        />);

        wrapper.simulate('click');
        const args = {
          type: 'num',
          value: 0,
        };

        expect(onClick).toHaveBeenCalledWith(args);
      });
    });
  });
});
