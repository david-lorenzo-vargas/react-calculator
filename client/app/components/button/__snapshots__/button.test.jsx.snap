// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render should render correctly 1`] = `
<button
  className="button button--theme-grey button--big"
  onClick={[Function]}
  type="button"
>
  0
</button>
`;

exports[`Button render should render correctly 2`] = `
<button
  className="button button--theme-grey"
  onClick={[Function]}
  type="button"
>
  0
</button>
`;

exports[`Button render should render correctly 3`] = `
<button
  className="button button--theme-grey button--pink"
  onClick={[Function]}
  type="button"
>
  0
</button>
`;
